com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ObjectStoreFileStorage.java

```diff
@@ -0,0 +1,186 @@
+    public int deleteFilesModifiedBefore(Date modificationTime) throws FileStorageException {
+        return removeBlobsByFilter(blob -> filterByModificationTime(blob, modificationTime));
+    }
+
+    @Override
+    public void processFileContent(FileDownloadProcessor fileDownloadProcessor) throws FileStorageException {
+        FileEntry fileEntry = fileDownloadProcessor.getFileEntry();
+        InputStream fileContentStream = null;
+        try {
+            Blob blob = blobStore.getBlob(container, fileEntry.getId());
+            Payload payload = blob.getPayload();
+            fileContentStream = payload.openStream();
+            fileDownloadProcessor.processContent(fileContentStream);
+        } catch (Exception e) {
+            throw new FileStorageException(e);
+        } finally {
+            if (fileContentStream != null) {
+                closeQuietly(fileContentStream);
+            }
+        }
+    }
+
+    private Map<String, String> createFileEntryMetadata(FileEntry fileEntry) {
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put(FileService.FileServiceColumnNames.SPACE.toLowerCase(), fileEntry.getSpace());
+        metadata.put(FileService.FileServiceColumnNames.FILE_NAME.toLowerCase(), fileEntry.getName());
+        metadata.put(FileService.FileServiceColumnNames.MODIFIED.toLowerCase(), Long.toString(fileEntry.getModified()
+            .getTime()));
+        if (fileEntry.getNamespace() != null) {
+            metadata.put(FileService.FileServiceColumnNames.NAMESPACE.toLowerCase(), fileEntry.getNamespace());
+        }
+        return metadata;
+    }
+
+    private int removeBlobsByFilter(Predicate<? super StorageMetadata> filter) {
+        Set<String> entriesToDelete = blobStore.list(container, new ListContainerOptions().withDetails())
+            .stream()
+            .filter(filter)
+            .map(blob -> blob.getName())
+            .collect(Collectors.toSet());
+
+        if (entriesToDelete.size() > 0) {
+            blobStore.removeBlobs(container, entriesToDelete);
+        }
+        return entriesToDelete.size();
+    }
+
+    private boolean filterByModificationTime(StorageMetadata blobMetadata, Date modificationTime) {
+        Map<String, String> userMetadata = blobMetadata.getUserMetadata();
+        if (userMetadata == null || userMetadata.isEmpty()) {
+            return true;
+        }
+        String longString = userMetadata.get(FileService.FileServiceColumnNames.MODIFIED.toLowerCase());
+        try {
+            long dateLong = Long.parseLong(longString);
+            Date date = new Date(dateLong);
+            return date.before(modificationTime);
+        } catch (NumberFormatException e) {
+            return true;
+        }
+    }
+
+    private boolean filterBySpace(StorageMetadata blobMetadata, String space) {
+        Map<String, String> userMetadata = blobMetadata.getUserMetadata();
+        if (userMetadata == null) {
+            return false;

```

Comment:
And why return false here?

Reply:
Because without the metadata we do not know the space/namespace. Such entries will be cleaned up by the cleanup.